src/Stripe.net/Services/ApplePayDomains/ApplePayDomainService.cs

```diff
@@ -21,44 +21,53 @@ public ApplePayDomainService(string apiKey)
-            return this.PostAsync($"{Urls.BaseUrl}/apple_pay/domains", requestOptions, cancellationToken, options);
+            return this.CreateEntityAsync(options, requestOptions, cancellationToken);
         }
 
         public virtual ApplePayDomain Delete(string domainId, RequestOptions requestOptions = null)
         {
-            return this.DeleteEntity($"{Urls.BaseUrl}/apple_pay/domains/{domainId}", requestOptions);
+            return this.DeleteEntityNew(domainId, null, requestOptions);
         }
 
         public virtual Task<ApplePayDomain> DeleteAsync(string domainId, RequestOptions requestOptions = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return this.DeleteEntityAsync($"{Urls.BaseUrl}/apple_pay/domains/{domainId}", requestOptions, cancellationToken);
+            return this.DeleteEntityAsyncNew(domainId, null, requestOptions, cancellationToken);
         }
 
         public virtual ApplePayDomain Get(string domainId, RequestOptions requestOptions = null)
         {
-            return this.GetEntity($"{Urls.BaseUrl}/apple_pay/domains/{domainId}", requestOptions);
+            return this.GetEntityNew(domainId, null, requestOptions);
         }
 
         public virtual Task<ApplePayDomain> GetAsync(string domainId, RequestOptions requestOptions = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return this.GetEntityAsync($"{Urls.BaseUrl}/apple_pay/domains/{domainId}", requestOptions, cancellationToken);
+            return this.GetEntityAsyncNew(domainId, null, requestOptions, cancellationToken);
         }
 
-        public virtual StripeList<ApplePayDomain> List(ApplePayDomainListOptions listOptions = null, RequestOptions requestOptions = null)
+        public virtual StripeList<ApplePayDomain> List(ApplePayDomainListOptions options = null, RequestOptions requestOptions = null)
         {
-            return this.GetEntityList($"{Urls.BaseUrl}/apple_pay/domains", requestOptions, listOptions);
+            return this.ListEntities(options, requestOptions);
         }
 
-        public virtual Task<StripeList<ApplePayDomain>> ListAsync(ApplePayDomainListOptions listOptions = null, RequestOptions requestOptions = null, CancellationToken cancellationToken = default(CancellationToken))
+        public virtual Task<StripeList<ApplePayDomain>> ListAsync(ApplePayDomainListOptions options = null, RequestOptions requestOptions = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return this.GetEntityListAsync($"{Urls.BaseUrl}/apple_pay/domains", requestOptions, cancellationToken, listOptions);
+            return this.ListEntitiesAsync(options, requestOptions, cancellationToken);
+        }
+
+        protected override string ClassUrl(string baseUrl = null)
+        {
+            baseUrl = baseUrl ?? StripeConfiguration.GetApiBase();
+
+            return $"{baseUrl}/apple_pay/domains";
```

Comment:
Not a fan of that override. I would rather calculate the relative path instead. Can we do "ObjectPath" or something instead?

Reply:
Yeah, that makes way more sense for a service actually. I changed `ObjectName` to `BasePath`.